app/redux/store.js

```diff
@@ -31,28 +31,31 @@ const zmqMiddlewareCreator = (zmq, subzmq) => {
 
       switch (type) {
         case 'CONNECT':
-          zmq.connect(`tcp://${sock.ip || 'localhost'}:5556`);
-          subzmq.connect(`tcp://${sock.ip || 'localhost'}:5555`);
+          reqzmq.connect('tcp://localhost:5556');
+          subzmq.connect('tcp://localhost:5555');
 
           break;
         case 'COMMAND':
-          if (sock.cmd !== "handshake") {
+          if (zmq.cmd && zmq.cmd.indexOf('start') === 0) {
             subzmq.subscribe('');
           }
-          zmq.send(sock.cmd);
+          if (zmq.cmd && zmq.cmd.indexOf('stop') === 0) {
+            subzmq.unsubscribe('');
+          }
+
+          reqzmq.send(zmq.cmd);
           break;
         case 'DISCONNECT':
-          subzmq.unsubscribe();
+          subzmq.unsubscribe('');
```

Comment:
Yes. There is a filter option for subscribing in zmq ( [code](https://github.com/JustinTulloss/zeromq.node/blob/master/lib/index.js#L478-L481) ). we need to use the same filter to unsubscribe as well.